lib/net.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Net - abstraction layer around Node's HTTP/S request system.
+ * Advantages:
+ * - easier acquiring of data
+ * - mass disabling of outgoing requests via Config.
+ */
+
+import * as https from 'https';
+import * as http from 'http';
+import * as url from 'url';
+import * as Streams from './streams';
+
+export class URIRequest {
+	uri: string;
+	statusCode?: number;
+	constructor(uri: string) {
+		this.uri = uri;
+		this.statusCode = undefined;
+	}
+	/**
+	 * Makes a basic http/https request to the URI.
+	 * Returns the response data.
+	 * @param opts request opts - headers, etc.
+	 */
+	async get(opts?: AnyObject): Promise<string | null> {
+		if (Config.noURIRequests) return null;
+		const protocol = url.parse(this.uri).protocol as string;
+		const net = protocol.includes('https:') ? https : http;
+		return new Promise((resolve) => {
+			const req = net.get(opts ? opts : this.uri, res => {
+				res.setEncoding('utf-8');
+				this.statusCode = res.statusCode;
+				void Streams.readAll(res).then(buffer => resolve(buffer));
+			});
+			req.on('error', (err) => {
+				throw err;
+			});
+			req.end();
+		});
```

Comment:
something something sync/async throws

```suggestion
		return new Promise((resolve, reject) => {
			const req = net.get(opts ? opts : this.uri, res => {
				res.setEncoding('utf-8');
				this.statusCode = res.statusCode;
				void Streams.readAll(res).then(buffer => resolve(buffer));
			});
			req.on('error', err => {
				reject(err);
			});
			req.end();
		});
```